src/search/move_picker.h

```diff
@@ -31,123 +35,147 @@ enum PickerPhase {
     Done,
 };
 
+template<movegen::MoveType moveType>
 class MovePicker {
 public:
-    MovePicker(
-        SearchTables& searchTables, uint8_t ply, std::optional<movegen::Move> ttMove = std::nullopt, std::optional<movegen::Move> prevMove = std::nullopt, PickerPhase phase = PickerPhase::TtMove)
+    MovePicker(SearchTables& searchTables, uint8_t ply, PickerPhase phase, std::optional<movegen::Move> ttMove = std::nullopt, std::optional<movegen::Move> prevMove = std::nullopt)
         : m_searchTables(searchTables)
         , m_ply(ply)
+        , m_phase(phase)
         , m_ttMove(ttMove)
         , m_prevMove(prevMove)
-        , m_phase(phase)
     {
     }
 
-    void setPhase(PickerPhase phase)
+    constexpr void resetPicking(PickerPhase phase)
     {
         m_phase = phase;
+        m_moves = movegen::ValidMoves();
+    }
+
+    constexpr uint16_t numGeneratedMoves()
+    {
+        return m_moves.count();
     }
 
-    template<Player player> constexpr std::optional<movegen::Move> pickNextMove(const BitBoard& board, movegen::ValidMoves& moves)
+    template<Player player> constexpr std::optional<movegen::Move> pickNextMove(const BitBoard& board)
     {
         switch (m_phase) {
+        case GenerateSyzygyMoves: {
+            if (syzygy::generateSyzygyMoves(board, m_moves)) {
+                m_phase = PickerPhase::Syzygy;
+            } else {
+                m_phase = PickerPhase::GenerateMoves;
+            }
```

Comment:
nitpick: I personally prefer explicit bools. If you don't agree just leave as is :)
```suggestion
            const bool syzygyActive = syzygy::generateSyzygyMoves(board, m_moves);
            if (syzygyActive) {
                m_phase = PickerPhase::Syzygy;
            } else {
                m_phase = PickerPhase::GenerateMoves;
            }
```